### PR TITLE
replace mandataris when updating state

### PR DESCRIPTION
**Pull request by Rahien**
replace mandataris when updating state, keeping selecting a replacement in the form for another story

image
image
image
image

### CODE REVIEW

File: app/components/mandatarissen/mandataris-card.js
Comment (bfidlers) [inline app/components/mandatarissen/mandataris-card.js:1]: These two functions are not used anymore I think?

File: app/components/mandatarissen/update-state.js [hunk 1]
```diff
@@ -134,11 +142,33 @@ export default class MandatarissenUpdateState extends Component {
...
+      const replacementMandataris =
+        await this.mandatarisService.getOrCreateReplacement(
+          this.args.mandataris,
+          this.selectedReplacement,
+          // passing these along because if we pass the model, relations will be
+          // evaluated as of right now and we haven't saved yet
+          newMandatarisProps,
+          this.selectedFractie
+        );
+      newMandataris.tijdelijkeVervangingen = [replacementMandataris];
```
Comment (bfidlers) [inline app/components/mandatarissen/update-state.js:162]: I'm a bit confused about this, it is possible to have multiple vervangingen, but we completely replace the array here?

File: app/components/mandatarissen/mandataris-card.hbs [hunk 2]
```diff
@@ -41,6 +41,51 @@
...
+                  ,
+                {{/unless}}
+              {{/if}}
+            {{/each}}
+          </div>
+        </div>
+      {{/if}}
+      {{#if @mandataris.tijdelijkeVervangingen}}
+        <div class="au-o-grid__item au-u-5-6">
+          <div class="au-c-description-label"><div>Vervanger van:</div></div>
```
Comment (bfidlers) [inline app/components/mandatarissen/mandataris-card.hbs:69]: I think this should be vervangen door

**Review by bfidlers — APPROVED** [review]
Few small remarks, otherwise looks good.

File: app/components/mandatarissen/mandataris-card.hbs [hunk 3]
```diff
@@ -41,6 +41,51 @@
...
+            {{#each @mandataris.vervangerVan as |replaced|}}
+              <AuLink
+                @route="mandatarissen.mandataris"
+                @model={{replaced.id}}
+                @skin="secondary"
+              >
+                {{replaced.isBestuurlijkeAliasVan.gebruikteVoornaam}}
+                {{replaced.isBestuurlijkeAliasVan.achternaam}}
+              </AuLink>
+              {{#if @last}}
```
[the remark this answers is not in the document]
Comment (Rahien) [reply app/components/mandatarissen/mandataris-card.hbs:58]: for posterity: you can't fixed in the next pr

File: app/components/mandatarissen/mandataris-card.hbs (hunk 2 above)
Comment (Rahien) [reply app/components/mandatarissen/mandataris-card.hbs:69]: fixed in the next pr

**Pull request merged**